src/platform/CYW30739/DiagnosticDataProviderImpl.cpp

```diff
@@ -124,6 +124,16 @@ void DiagnosticDataProviderImpl::ReleaseNetworkInterfaces(NetworkInterface * net
         Platform::Delete(del);
     }
 }
+
+CHIP_ERROR DiagnosticDataProviderImpl::ResetWatermarks()
+{
+    // If implemented, the server SHALL set the value of the CurrentHeapHighWatermark attribute to the
+    // value of the CurrentHeapUsed.
+    // On CYW30739, overide with non-op to pass CI.
```

Comment:
This does not make sense.  CI should not be assuming this is supported, if this provider says it's not supported: this is an optional feature.  What CI is this trying to pass?